src/com/amazon/corretto/crypto/provider/EvpHpkePrivateKey.java

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import com.amazon.corretto.crypto.provider.EvpKey.CanDerivePublicKey;
+import java.security.PrivateKey;
+
+public class EvpHpkePrivateKey extends EvpHpkeKey
+    implements PrivateKey, CanDerivePublicKey<EvpHpkePublicKey> {
+  private static final long serialVersionUID = 1;
+
+  EvpHpkePrivateKey(InternalKey key, HpkeParameterSpec spec) {
+    super(key, false, spec);
+  }
+
+  EvpHpkePrivateKey(final long ptr, HpkeParameterSpec spec) {
+    this(new InternalKey(ptr), spec);
+  }
+
+  // Copied from EvpEcPrivateKey
```

Comment:
You can drop this comment, the provenance isn't necessary for something this small.